File: src/main/java/com/github/davinkevin/transmissionrss/feeds/properties/FeedsProperty.java
package com.github.davinkevin.transmissionrss.feeds.properties;

import com.github.davinkevin.transmissionrss.feeds.model.FeedProperty;
import io.vavr.collection.List;
import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * Created by kevin on 08/04/2018
 */
@Configuration
@ConfigurationProperties
public class FeedsProperty {
    @Getter List<FeedProperty> feeds;

    public void setFeeds(java.util.List<FeedProperty> feedProperties) {
        this.feeds = List.ofAll(feedProperties);
    }
}

File: src/main/kotlin/com/github/davinkevin/transmissionrss/feeds/properties/FeedsProperty.kt
package com.github.davinkevin.transmissionrss.feeds.properties

import com.github.davinkevin.transmissionrss.feeds.model.FeedProperty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

/**
 * Created by kevin on 08/04/2018
 */
@Configuration
@ConfigurationProperties
data class FeedsProperty (
    var feeds: List<FeedProperty> = ArrayList()
)
